### PR TITLE
Update Dependencies: Wagmi, rainbowkit, and their dependencies

### PR DESCRIPTION
**Pull request by ghimiresdp**
- feat: added alchemy provider instead of public
- feat(package): updated dependencies
- refactor(ui): wagmi migrations in components
- feat(UI): updated Rainbowkit, Wagmi, and its implementations and removed SSR issues
- fix: removed unused imports
- fix(faucet): migrate faucet methods to the latest version of wagmi
- fix: update eslint and prettier versions
- fix: update wagmi changes to Answersheet
- fix: update Chakra UI versions

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -1,63 +1,39 @@
...
-const wagmiClient = createClient({
-  autoConnect: true,
-  connectors: connectors,
-  provider,
+import { getDefaultConfig } from '@rainbow-me/rainbowkit';
+import { RainbowKitProvider } from '@rainbow-me/rainbowkit';
+
+const config = getDefaultConfig({
+  appName: 'cit-web3-class',
+  projectId: '1b5b1d5c351c838062b62002e37bee3c',
```
Comment (HokutoMorita) [inline src/pages/_app.tsx:14]: Is the projectId value fixed?
Also, what do you need it for?

**Comment by HokutoMorita** [comment]
@ghimiresdp 
Thank you !
I have confirmed that the Polygon Mumbai Error and Next.js render error have been resolved.

File: src/components/wallet/ContractDetail.tsx [hunk 2]
```diff
@@ -11,15 +14,15 @@ import {
...
 } from '@chakra-ui/react';
 import useTranslation from 'next-translate/useTranslation';
-import { useAccount, useNetwork } from 'wagmi';
+import { useAccount } from 'wagmi';
 import { defaultChain, getContractAddress } from '@/utils/contract';
 import { useTokenBalance } from '@/hooks/useTokenBalance';
 
 export const ContractDetail = () => {
   const { t } = useTranslation('common');
   const { address, connector, isConnected } = useAccount();
```
Comment (HokutoMorita) [inline src/components/wallet/ContractDetail.tsx:23]: useAccount() is declared on lines 23 and 25.

23 lines
```
const { address, connector, isConnected } = useAccount();
```

25lines
```
const { chain } = useAccount();
```

Comment (HokutoMorita) [reply src/components/wallet/ContractDetail.tsx:23]: Also, the `connector` variable declared on line 23 is not used.

**Comment by HokutoMorita** [comment]
An error has occurred in cit-coin-ui/src/hooks/useTokenBalance.ts.

First:
```
'useNetwork' が宣言されていますが、その値が読み取られることはありません。ts(6133)
モジュール '"wagmi"' にエクスポートされたメンバー 'useNetwork' がありません。ts(2305
```

スクリーンショット 2024-03-12 14 21 55

**Comment by ghimiresdp** [comment]
@HokutoMorita 
I see, I will fix it and push new commits for the fixes

**Comment by HokutoMorita** [comment]
Thank you !
I haven't been able to review any other files yet.
So I might make some other comments as well.

File: src/components/NFT.tsx [hunk 3]
```diff
@@ -54,14 +62,12 @@ const AssetCard = (props: { asset: Asset }) => {
...
 
-
 export const AssetLibrary = () => {
-  const { chain } = useNetwork();
   const [assets, setAssets] = useState<Asset[]>([]);
   const { isOpen, onOpen, onClose } = useDisclosure();
   const { t } = useTranslation('common');
   const [loading, setLoading] = useState(false);
-  const { address, connector, isConnected } = useAccount();
+  const { address, connector, isConnected, chain } = useAccount();
```
Comment (HokutoMorita) [inline src/components/NFT.tsx:70]: Also, the `connector` and `isConnected` variables declared on line 23 are not used.

**Review by HokutoMorita — CHANGES_REQUESTED** [review]

**Comment by HokutoMorita** [comment]
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/components/NFT.tsx`

I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
スクリーンショット 2024-03-12 14 52 16
スクリーンショット 2024-03-12 14 52 07

TypeScript type definition may be wrong. ?...

**Comment by HokutoMorita** [comment]
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/hooks/useContractConfig.ts`
```
'useNetwork' が宣言されていますが、その値が読み取られることはありません。
```
スクリーンショット 2024-03-12 14 58 21
`useNetwork` may not be used

**Comment by HokutoMorita** [comment]
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/components/Answersheet.tsx`

77 lines
I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
スクリーンショット 2024-03-12 15 11 22

180 lines
I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
スクリーンショット 2024-03-12 15 12 09

**Comment by HokutoMorita** [comment]
@ghimiresdp 
An error has occurred in `cit-coin-ui/src/pages/faucet.tsx`

49 lines
I'm getting the error "0 arguments expected, 1 specified."
```
0 個の引数が必要ですが、1 個指定されました。
```
スクリーンショット 2024-03-12 15 15 00


57 lines
I'm getting the error "The type of parameter 'd' is implicitly 'any'."
```
パラメーター 'd' の型は暗黙的に 'any' になります。
```
スクリーンショット 2024-03-12 15 16 37

**Comment by HokutoMorita** [comment]
@ghimiresdp 
There is something additional I found out about this comment. !
https://github.com/henkaku-center/cit-coin-ui/pull/9#issuecomment-1990534346

The `watch` argument for wagmi@2.5.7's useBalance hook may be remove.

v1.x wagmi
https://1.x.wagmi.sh/react/hooks/useBalance
v1.x has a `watch` argument.

v2.x wagmi
https://wagmi.sh/react/api/hooks/useBalance
v2.x does not have a `watch` argument.

File: src/pages/_app.tsx (hunk 1 above)
Comment (ghimiresdp) [reply src/pages/_app.tsx:14]: @HokutoMorita 

I am so sorry for the late reply,
the `Rainbow Kit` library needs project ID from `WalletConnect` for integration. The project ID is fixed for a DAPP so we can keep as it is in the source code.

**Comment by ghimiresdp** [comment]
@HokutoMorita 
I have fixed some issues you previously mentioned. Most of the problems were fixed once I deleted node modules and re ran `yarn` command. can you try it once?

File: src/pages/_app.tsx (hunk 1 above)
Comment (HokutoMorita) [reply src/pages/_app.tsx:14]: I see
Thank you!

**Comment by HokutoMorita** [comment]
> @HokutoMorita I have fixed some issues you previously mentioned. Most of the problems were fixed once I deleted node modules and re ran `yarn` command. can you try it once?

@ghimiresdp 
Please show me your VScode screenshot !

For Example
`cit-coin-ui/src/components/NFT.tsx`
スクリーンショット 2024-03-15 14 31 23


> Most of the problems were fixed once I deleted node modules and re ran yarn command. can you try it once?

I've already tried this method.
But the error remains.

**Comment by ghimiresdp** [comment]
@HokutoMorita 
There seems to be no problem in my case

I think you might need to change typescript settings in your VS Code
`settings.json`
```diff
{
-     "typescript.enablePromptUseWorkspaceTsdk": true,
+     // "typescript.enablePromptUseWorkspaceTsdk": true,
}
```

image
image

**Review by HokutoMorita — APPROVED** [review]
LGTM！！！

Thank you !!

**Pull request merged**